### ISSUE TITLE
In imageset have different methods to differentiate between static/dynamic masking

### ISSUE THREAD

**Issue by jmp1985** [dials/dials#474]
Masking can come from different locations (e.g. external file, trusted range, panel untrusted rectangle and dynamic shadowing) Some of these are static across the (external mask), others can be generated on the fly trivially from the panel and image data (trusted range, panel untrusted rectangle) and others are more complex (dynamic shadowing).  

In the parallel integration, we want to load the image data then process reflections in parallel, this involves loading a block of images into memory and then for each reflection, extracting the shoebox, processing, then dumping. If the mask is static or trivial to compute then the mask for each shoebox can be computed on the fly. However, if  true dynamic masking is done (e.g. the work Richard did to model the shadow of the goniometer for I23) then it is probably more efficient to allocate another block for the mask. Trade off is memory vs cpu. 

In order to make this efficient in both cases, I need to know if the mask is trivial or dynamic. Having some method (dynamic_masking() == True) would be useful. Also having get_mask(), get_dynamic_mask() methods. Or something like this. Also packaging dynamic masking into a C++ callable object might be useful for this maybe.

**Comment by jmp1985** [comment]
This is already done.
